### DISCUSSION TITLE
Using a public certificate(s) generated by an IDP such as Okta or Azure (or using the JWKS url) in order to validate tokens

### DISCUSSION THREAD

**Discussion by KellerKev** [caprover/caprover#1733] [Q&A] [answered]
I would like to replace the public key that is used in caprover with a public key set from another identity provider. 
Now when I get the jwt token from e.g. okta or azure I can use that to login to caprover as caprover will be able to verifiy the signature with the new keys. 

Could you help find the key material and where caprover generates or hosts it and how I can intercept in this process to place some other public key materterial please? 

This way customers can take advantage of okta & azure etc. SSO style token aquiring and then use them to login to caprover.

**Comment by githubsaturn** [comment]
> the public key that is used in caprover

What public key?

**Comment by KellerKev** [comment]
These are for example thr public keys required to verify an jwt token coming from Azure https://login.microsoftonline.com/common/discovery/v2.0/keys

**Reply by githubsaturn** [reply]
CapRover doesn't use public key for jwt tokens, it uses a secret:
https://github.com/caprover/caprover/blob/0e8656213da1bbba1b8cc5ecdde93416924790ec/src/user/Authenticator.ts#L112-L118

**Reply by githubsaturn** [reply, marked as answer]
Secret is a UUID stored as a [docker swarm secret](https://docs.docker.com/engine/swarm/secrets/).
